src/aave-v3/SupplyVault.sol

```diff
@@ -225,25 +224,27 @@ contract SupplyVault is ISupplyVault, SupplyVaultBase {
         }
     }
 
-    function _getUnaccruedReward(
+    function _getUpdatedUnclaimedReward(
         address _user,
         address _rewardToken,
         uint256 _claimableReward,
         uint256 _totalSupply
-    ) internal view returns (uint128 unaccruedReward) {
-        unaccruedReward = _getUnaccruedRewardFromUnaccruedRewardIndex(
-            balanceOf(_user),
-            _getUnaccruedRewardIndex(_claimableReward, _totalSupply) + // The unaccrued reward index
-                rewardsIndex[_rewardToken] -
-                userRewards[_rewardToken][_user].index // The difference between the current reward index and the user's index
-        );
+    ) internal view returns (uint128 unclaimed) {
+        unclaimed =
+            userRewards[_rewardToken][_user].unclaimed +
```

Comment:
`userRewards[_rewardToken][_user]` could be copied to memory to potentially save the cost of reading from hot storage slot minus the cost of reading from hot memory slot